src/Sonarr.Api.V3/Qualities/QualityDefinitionController.cs

```diff
@@ -20,6 +20,15 @@ public QualityDefinitionController(IQualityDefinitionService qualityDefinitionSe
             : base(signalRBroadcaster)
         {
             _qualityDefinitionService = qualityDefinitionService;
+
+            SetupValidation(qualityDefinitionService);
+        }
+
+        private void SetupValidation(IQualityDefinitionService qualityDefinitionService)
+        {
+            var limits = qualityDefinitionService.GetLimits();
```

Comment:
This is what I was talking about before, you need to make an instance of `QualityDefinitionLimits` to use in the validator instead of just accessing the limits from a static class.

Since `QualityDefinitionResourceValidator` will never actually use different limits within the same version of the application it makes sense to use a static value instead of initializing with a dynamic value.

Reply:
Is creating that instance actually a problem here, though? In the grander scheme of things, I don't think it really matters. Please remember my original points as to why these members are not static. This isn't me just being hard-headed about it.

I completely understand what you're saying, but on the other hand, if you look at this objectively it really doesn't matter if they're static or not from a performance standpoint. There's just not enough overhead here for it to matter, IMO.

In our previous discussion, I did clarify with you whether or not you wanted to duplicate the structure (once for serialization and once as more of a domain object) but it seemed like you agreed at the time that the duplication (maintenance overhead) wasn't worth it, which I agree with. But if you decide to change your mind on that, it may open the opportunity to use static values here.

I'm still 100% ok with this code given:

1. The points I made in our earlier discussion about why they are not static
2. The performance overhead / difference is negligible.
3. Code can always be changed and refactored later should the first two points ever change.

I'll go ahead and resolve this conversation for now for the sake of making progress and getting things merged. This PR has already been open quite a while and I want to try to come to a compromise as much as possible to avoid this turning into something akin to the level of paid work.

Reply:
I just saw your other comment on `QualityDefinitionLimits.cs` (but I can't comment on that directly, for some reason...)

If you're OK with the two separate classes I will go ahead and make that change out of respect for your request, although I personally do not see an issue here. Hopefully after this we can get this merged without much fuss.

Reply:
This is what I was thinking: https://github.com/Sonarr/Sonarr/commit/954903812197d0fad3bc091e6e469342680f7776

Reply:
So basically you're OK with no separation of concerns between the controller and the logic. Not that there's much logic there to begin with, but I'm more interested in the principles in this case. I'll go ahead and adjust my code to remove logic from the service class.

Reply:
Resolved in latest push